solr/docker/build.gradle

```diff
@@ -18,106 +18,192 @@
 import com.google.common.base.Preconditions
 import com.google.common.base.Strings
 
-apply plugin: 'base'
-apply plugin: 'com.palantir.docker'
-
-subprojects {
-  apply plugin: 'base'
-  apply plugin: 'com.palantir.docker'
-}
-
 description = 'Solr Docker image'
 
-def dockerPackage = project(':solr:docker:package')
-
-dependencies {
-  docker dockerPackage
-}
+apply plugin: 'base'
 
+// Solr Docker inputs
 def dockerImageRepo = propertyOrEnvOrDefault("solr.docker.imageRepo", "SOLR_DOCKER_IMAGE_REPO", "apache/solr")
 def dockerImageTag = propertyOrEnvOrDefault("solr.docker.imageTag", "SOLR_DOCKER_IMAGE_TAG", "${version}")
 def dockerImageName = propertyOrEnvOrDefault("solr.docker.imageName", "SOLR_DOCKER_IMAGE_NAME", "${dockerImageRepo}:${dockerImageTag}")
 def baseDockerImage = propertyOrEnvOrDefault("solr.docker.baseImage", "SOLR_DOCKER_BASE_IMAGE", 'openjdk:11-jre-slim')
 def githubUrlOrMirror = propertyOrEnvOrDefault("solr.docker.githubUrl", "SOLR_DOCKER_GITHUB_URL", 'github.com')
 
-docker {
-  name = dockerImageName
-  files file('include')
-  buildArgs(['BASE_IMAGE' : baseDockerImage, 'SOLR_PACKAGE_IMAGE' : 'apache/solr-build:local-package', 'SOLR_VERSION': "${version}", 'GITHUB_URL': githubUrlOrMirror])
+// Build directory locations
+def dockerBuildDistribution = "$buildDir/distributions"
+def imageIdFile = "$buildDir/image-id"
+
+configurations {
+  packaging {
+    canBeResolved = true
+  }
+  dockerImage {
+    canBeResolved = true
+  }
+}
+
+dependencies {
+  packaging project(path: ":solr:packaging", configuration: 'archives')
+
+  dockerImage files(imageIdFile) {
+    builtBy 'dockerBuild'
+  }
 }
 
-tasks.docker {
-  // In order to create the solr docker image, the solr package image must be created first.
-  dependsOn(dockerPackage.tasks.docker)
+task dockerTar(type: Tar) {
```

Comment:
I bet this task could be skipped entirely in lieu of the packaging tgz directly being provided as stdin to Docker.  If it doesn't contain these scripts, we could arrange for that easily enough.  I'm toying with the Dockerfile right now that could operate on this assumption.  It's eliminate a tar.gz generation and decompression, and consequently overall simpler.

Reply:
Oops; I'm mistaken, as it must contain the Dockerfile at the root.

Reply:
Yeah it'd be nice if the build via TGZ were a bit more flexible.

Reply:
In our container centric world right now... I think it would not be a big deal if our tgz included a Dockerfile at the root.  I'm not sure I want to push for that at the moment (would others hate it?), but throwing it out there.  Re-building *any* official tgz our project produces would be marvelously simple then :-). (curl some official tgz then pipe right into docker build :-)

Reply:
I don't think it's a bad idea. Possibly an effort to push for separately. Might be an issue if we start auto-loading first-party-packages in the docker image in the future. I imagine that those wouldn't be included in the solr tgz, but I might be wrong about that.

Reply:
The plan for 1P packages does not conflict with this.  They remain where they are, and eventually might only ship metadata in the image instead of all the JARs which would be fetched on demand.  I don't think there's a Docker impact to that.

Reply:
@dsmiley I still think that your idea is something that we should explore. I think that this PR (that I just merged) is just a step in the right direction. But we might as well have something that is working much better now, and add final touches later.